src/HDF5.jl

```diff
@@ -958,35 +1008,27 @@ function setindex!(p::HDF5Properties, val, name::String)
 end
 # Create a dataset with properties: obj[path, prop1, set1, ...] = val
 function setindex!(parent::Union{HDF5File, HDF5Group}, val, path::String, prop1::String, val1, pv...)
-    if !iseven(length(pv))
-        error("Properties and values must come in pairs")
+    t = (prop1, val1, pv...)
+    checkprops(t...)
+    dcpl = p_create(H5P_DATASET_CREATE)
+    dxpl = p_create(H5P_DATASET_XFER, false, t...)
+    dapl = p_create(H5P_DATASET_ACCESS, false, t...)
+
+    need_chunks = any(t[i] in chunked_props for i=1:2:length(t))
+    have_chunks = any(t[i] == "chunk"       for i=1:2:length(t))
+    chunk = need_chunks ? heuristic_chunk(val) : Int[]
+    discard_chunks = need_chunks && isempty(chunk)
+    if need_chunks && !have_chunks && !discard_chunks
+        dcpl["chunk"] = chunk
```

Comment:
is it not possible to remove chunk on line 1019
and just put 
`dcpl["chunk"] = chunk`
I realize in the old code `chunk = heuristic_chunk(val)` always?

Reply:
We cannot just set `dcpl`, because it could a) be already preset by the user and b) we need to ensure that chunk is not empty (`heuristic_chunk` can return empty chunks as well).